src/ui/mainwindow.cc

```diff
@@ -233,8 +233,9 @@ MainWindow::MainWindow( Config::Class & cfg_ ):
   navPronounce->setEnabled( false );
   navToolbar->widgetForAction( navPronounce )->setObjectName( "soundButton" );
 
-  connect( navPronounce, SIGNAL( triggered() ),
-           this, SLOT( pronounce() ) );
+  connect( navPronounce, &QAction::triggered, [ this ]() {
+    getCurrentArticleView()->playSound();
+  } );
```

Comment:
will getCurrentArticleView() return null?

Reply:
No, because the button is disabled if there is no CurrentArticleView.

In the original code, the `pronounce` will just go straight to `getCurrentArticleView`, so the new code is identical to the original but without the little overhead of type mismatch.